### ISSUE TITLE
kube-controller-manager never gives up trying to create invalid LB

### ISSUE THREAD

**Issue by zmerlynn** [kubernetes/kubernetes#20855]
Simple repo:

Try to create this invalid service, `badsvc.json`:

```
{
  "kind": "Service",
  "apiVersion": "v1",
  "metadata": {
    "name": "test",
    "labels": {
      "app": "test"
    }
  },
  "spec": {
    "ports": [
      {
        "port": 80,
        "targetPort": "http"
      }
    ],
    "type": "LoadBalancer",
    "loadBalancerIP": "8.8.8.8"
  }
}
```

```
kubectl create -f /tmp/badsvc.json
kubectl get events --all-namespaces -w # Wait for errors
kubectl delete svc/test
kubectl get events --all-namespaces -w # <-- There are still errors
```

**Comment by zmerlynn** [comment]
Sample events from 1.2 cluster:
```
default   3m        0s        40        test      Service             CreatingLoadBalancer   {service-controller }   Creating load balancer
default   3m        0s        40        test      Service             CreatingLoadBalancerFailed   {service-controller }   Error creating load balancer (will retry): Failed to create load balancer for service default/test: requested ip 8.8.8.8 is neither static nor assigned to LB acdb2b00aceb011e5857642010af0001: <nil>
```
(ad infinitum)

**Comment by freehan** [comment]
ref: #21952

**Comment by xiangpengzhao** [comment]
As per discussion in #21952, this should be resolved by #22069.

/cc @thockin
